packages/vite/src/node/preview.ts

```diff
@@ -97,20 +98,35 @@ export async function preview(
 
   app.use(compression())
 
+  const { spa } = config
+
   // static assets
   const distDir = path.resolve(config.root, config.build.outDir)
   app.use(
     config.base,
     sirv(distDir, {
       etag: true,
       dev: true,
-      single: true
+      single: spa
     })
   )
 
   // apply post server hooks from plugins
   postHooks.forEach((fn) => fn && fn())
 
+  if (!spa) {
```

Comment:
I'm not entirely sure we should add this block. It seems a bit too specific to a single use case. E.g. SvelteKit targets many different platforms and not all of them will have this behavior. I think it'd be something that would be easy to add in a middleware rather than adding it here

If we were to keep it, do we actually need the `if (!spa)` check here?

Reply:
Makes sense. Done.